src/com/android/launcher3/util/PackageManagerHelper.java

```diff
@@ -256,21 +256,34 @@ public static IntentFilter getPackageFilter(String pkg, String... actions) {
         return packageFilter;
     }
 
+    public static boolean isSystemApp(Context context, String pkgName) {
+        return isSystemApp(context, null, pkgName);
+    }
+
     public static boolean isSystemApp(Context context, Intent intent) {
+        return isSystemApp(context, intent, null);
+    }
+
+    public static boolean isSystemApp(Context context, Intent intent, String pkgName) {
```

Comment:
It would be better if there were only 2 functions:

- Remove `isSystemApp(Context context, String pkgName)`
- Change `isSystemApp(Context context, Intent intent, String pkgName)` to only take `Context` and `String` package name.
- Modify `isSystemApp(Context context, Intent intent)` so it converts `Intent` to `String` package name internally and just passes it to the *main* function.

(do not apply the suggestion automatically/on web, I could not select the whole block it should replace)

```suggestion
        public static boolean isSystemApp(Context context, Intent intent) {
        PackageManager pm = context.getPackageManager();
        // Get the package name for intent
        String packageName = null;
        if (intent != null) {
            ComponentName cn = intent.getComponent();
            if (cn == null) {
                ResolveInfo info = pm.resolveActivity(intent, PackageManager.MATCH_DEFAULT_ONLY);
                if ((info != null) && (info.activityInfo != null)) {
                    packageName = info.activityInfo.packageName;
                }
            } else {
                packageName = cn.getPackageName();
            }
        }
        return isSystemApp(context, packageName);
    }

    public static boolean isSystemApp(Context context, String packageName) {
        PackageManager pm = context.getPackageManager();
        // Check if the provided package is a system app.
        if (packageName != null) {
            try {
                PackageInfo info = pm.getPackageInfo(packageName, 0);
                return (info != null) && (info.applicationInfo != null) &&
                        ((info.applicationInfo.flags & ApplicationInfo.FLAG_SYSTEM) != 0);
            } catch (NameNotFoundException e) {
                return false;
            }
        } else {
            return false;
        }
    }
```